### DISCUSSION TITLE
DLL for php 8.2.x under windows

### DISCUSSION THREAD

**Discussion by dontaskdonttellme** [phpredis/phpredis#2455] [General]
When I try to locate the DLL for php 8.2.x under windows, it shows the 8.2 DLL as expired and is not able to be downloaded.  Can someone please fix this so we can download it?

**Comment by firewizard** [comment]
Is there a plan to release php 8.2 support for win x86/x64?

**Reply by michael-grunder** [reply]
Disclaimer:  I haven't had a Windows machine in many years, but we do build these DLLs in actions.

[There are Windows artifacts here at if you scroll down](https://github.com/phpredis/phpredis/actions/runs/8423459450).

[PHP 8.2 NTS](https://github.com/phpredis/phpredis/actions/runs/8423459450/artifacts/1356365354)
[PHP 8.2 TS](https://github.com/phpredis/phpredis/actions/runs/8423459450/artifacts/1356365356)

Do those work or do we need to build them differently.

**Reply by firewizard** [reply]
I tried using those artifacts and it didn't work, that's why I thought it wasn't the correct approach. Tried it again with the links you provided, downloaded the nts version and got this when I tried to load the extension:

`PHP Warning:  PHP Startup: Unable to load dynamic library 'redis' (tried: d:/www/php8.2/ext\redis (The specified module could not be found), d:/www/php8.2/ext\php_redis.dll (%1 is not a valid Win32 application)) in Unknown on line 0`

php version is `PHP 8.2.17 (cli) (built: Mar 12 2024 16:03:40) (NTS Visual C++ 2019 x86)`

**Reply by michael-grunder** [reply]
Ahh, OK.  In that case I'm not sure.

I can look into it but I don't have a Windows machine to test with.

**Comment by firewizard** [comment]
I can test the dlls any time, but I can't help you with compiling them...

**Reply by firewizard** [reply]
@michael-grunder I switched to the 64 bit version of php 8.2 and the extension worked fine. thanks for the help, much appreciated!
